<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>CiteFuse • BuildABiocWorkshop2020</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="CiteFuse">
<meta property="og:description" content="BuildABiocWorkshop2020">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BuildABiocWorkshop2020</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/CiteFuse_BioCAsia_workshop.html">CiteFuse</a>
    </li>
    <li>
      <a href="../articles/HOWTO_BUILD_WORKSHOP.html">How To Build A Workshop Package</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/seandavi/BuildABiocWorkshop2020">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="CiteFuse_BioCAsia_workshop_files/htmlwidgets-1.5.1/htmlwidgets.js"></script><script src="CiteFuse_BioCAsia_workshop_files/jquery-1.12.4/jquery.min.js"></script><link href="CiteFuse_BioCAsia_workshop_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="CiteFuse_BioCAsia_workshop_files/datatables-binding-0.14/datatables.js"></script><link href="CiteFuse_BioCAsia_workshop_files/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="CiteFuse_BioCAsia_workshop_files/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="CiteFuse_BioCAsia_workshop_files/dt-core-1.10.20/js/jquery.dataTables.min.js"></script><link href="CiteFuse_BioCAsia_workshop_files/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet">
<script src="CiteFuse_BioCAsia_workshop_files/crosstalk-1.1.0.1/js/crosstalk.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>CiteFuse</h1>
                        <h4 class="author">Yingxin Lin and Hani Jieun Kim<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>
</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/seandavi/BuildABiocWorkshop2020/blob/master/vignettes/CiteFuse_BioCAsia_workshop.Rmd"><code>vignettes/CiteFuse_BioCAsia_workshop.Rmd</code></a></small>
      <div class="hidden name"><code>CiteFuse_BioCAsia_workshop.Rmd</code></div>

    </div>

    
    
<div id="workshop-title" class="section level1">
<h1 class="hasAnchor">
<a href="#workshop-title" class="anchor"></a>Workshop Title</h1>
<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<div id="instructors-and-contact-information" class="section level3">
<h3 class="hasAnchor">
<a href="#instructors-and-contact-information" class="anchor"></a>Instructors and contact information</h3>
<ul>
<li>Yingxin Lin (<a href="mailto:yingxin.lin@sydney.edu.au" class="email">yingxin.lin@sydney.edu.au</a>)</li>
<li>Hani Jieun Kim (<a href="mailto:hani.kim@sydney.edu.au" class="email">hani.kim@sydney.edu.au</a>)</li>
</ul>
</div>
<div id="description" class="section level3">
<h3 class="hasAnchor">
<a href="#description" class="anchor"></a>Description</h3>
<p>The latest breakthrough in single-cell omics is on multi-modal profiling of different biomolecule species in individual cells. Among the fast evolving biotechnologies developed for multi-modal profiling, cellular indexing of transcriptomes and epitopes by sequencing (CITE-seq) is attracting attention, especially in the field of immunology, given its ability to simultaneously quantify global gene expression and cellular proteins using RNA-sequencing and antibody-derived tags (ADTs) on single cells. While the additional protein marker information brought about by ADTs is extremely valuable, new biological insights can only be gained by developing analytic methods that fully take advantage of the complementarity between mRNA and ADT expression measured in CITE-seq.</p>
<p>To address this, we developed a streamlined pipeline–CiteFuse–that consists of a suite of tools for the integration and the downstream analysis of CITE-seq data. In this workshop, we will provide a hands-on experience to the CiteFuse package and cover all the steps including doublet detection, modality integration, cell type clustering, differential RNA and ADT expression analysis, ADT evaluation, and ligand–receptor interaction analysis on a publicly available CITE-seq dataset. We also demonstrate the applicability of CiteFuse package on other multi-modal data types by applying our pipeline on the recently developed ASAP-seq data.</p>
<p><a href="https://sydneybiox.github.io/CiteFuse/articles/CiteFuse.html">This vignette</a> provides a more complete description of the various tools in CiteFuse and will serve as the basis of our workshop.</p>
</div>
<div id="pre-requisites" class="section level3">
<h3 class="hasAnchor">
<a href="#pre-requisites" class="anchor"></a>Pre-requisites</h3>
<p>Software:</p>
<ul>
<li>Basic knowledge of R syntax</li>
<li>Familiarity with single-cell RNA-sequencing</li>
<li>Familiarity with the <code>SingleCellExperiment</code> class</li>
</ul>
<p>Background reading:</p>
<ul>
<li>The textbook “Orchestrating Single-Cell Analysis with Bioconductor” is a great reference for single-cell analysis using Bioconductor packages.</li>
<li><a href="https://academic.oup.com/bioinformatics/article-abstract/36/14/4137/5827474?redirectedFrom=fulltext">CiteFuse enables multi-modal analysis of CITE-seq data</a></li>
<li><a href="https://www.nature.com/articles/nmeth.4380">Simultaneous epitope and transcriptome measurement in single cells</a></li>
</ul>
</div>
<div id="participation" class="section level3">
<h3 class="hasAnchor">
<a href="#participation" class="anchor"></a>Participation</h3>
<p>Participants will learn how to…</p>
</div>
<div id="r-bioconductor-packages-used" class="section level3">
<h3 class="hasAnchor">
<a href="#r-bioconductor-packages-used" class="anchor"></a><em>R</em> / <em>Bioconductor</em> packages used</h3>
<ul>
<li>This workshop will focus on Bioconductor packages [SingleCellExperiment] (<a href="https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html" class="uri">https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html</a>) and <a href="https://academic.oup.com/bioinformatics/article-abstract/36/14/4137/5827474?redirectedFrom=fulltext">CiteFuse</a>.</li>
</ul>
</div>
<div id="time-outline" class="section level3">
<h3 class="hasAnchor">
<a href="#time-outline" class="anchor"></a>Time outline</h3>
<p>An example for a 55-minute workshop:</p>
<table class="table">
<thead><tr class="header">
<th>Activity</th>
<th>Time</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Introduction</td>
<td>15m</td>
</tr>
<tr class="even">
<td>Data processing and integration</td>
<td>15m</td>
</tr>
<tr class="odd">
<td>Downstream analysis</td>
<td>15m</td>
</tr>
<tr class="even">
<td>Application of CiteFuse on ASAP-seq</td>
<td>5m</td>
</tr>
<tr class="odd">
<td>Wrap-up and Conclusions</td>
<td>5m</td>
</tr>
</tbody>
</table>
</div>
<div id="workshop-goals-and-objectives" class="section level3">
<h3 class="hasAnchor">
<a href="#workshop-goals-and-objectives" class="anchor"></a>Workshop goals and objectives</h3>
<p>List “big picture” student-centered workshop goals and learning objectives. Learning goals and objectives are related, but not the same thing. These goals and objectives will help some people to decide whether to attend the conference for training purposes, so please make these as precise and accurate as possible.</p>
<p><em>Learning goals</em> are high-level descriptions of what participants will learn and be able to do after the workshop is over. <em>Learning objectives</em>, on the other hand, describe in very specific and measurable terms specific skills or knowledge attained. The <a href="#bloom">Bloom’s Taxonomy</a> may be a useful framework for defining and describing your goals and objectives, although there are others.</p>
</div>
<div id="learning-goals" class="section level3">
<h3 class="hasAnchor">
<a href="#learning-goals" class="anchor"></a>Learning goals</h3>
<p>Some examples:</p>
<ul>
<li>describe how to…</li>
<li>identify methods for…</li>
<li>understand the difference between…</li>
</ul>
</div>
<div id="learning-objectives" class="section level3">
<h3 class="hasAnchor">
<a href="#learning-objectives" class="anchor"></a>Learning objectives</h3>
<ul>
<li>analyze xyz data to produce…</li>
<li>create xyz plots</li>
<li>evaluate xyz data for artifacts</li>
</ul>
</div>
</div>
</div>
<div id="analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#analysis" class="anchor"></a>Analysis</h1>
<div id="the-dataset" class="section level2">
<h2 class="hasAnchor">
<a href="#the-dataset" class="anchor"></a>The dataset</h2>
<p>The dataset we will be working with concerns a single-cell RNA-sequencing dataset consisting of two different experiments, which correspond to two treatments. <a href="https://www.nature.com/articles/s41588-019-0489-5">McFaline-Figueroa et al.</a> studied the epithelial-to-mesenchymal transition (EMT), where cells spatially migrate from the epithelium to the mesenchyme during development. This process will be described by a trajectory, reflecting the gene expression changes occurring during this migration. The authors furthermore studied both a control (<code>Mock</code>) condition, and a condition in the absence of transforming growth factor <span class="math inline">\(\beta\)</span> (TGFB).</p>
<p>In summary, we will be predominantly investigating a CITE-seq dataset consisting of PBMCs. For simplicity, this lineage is studied in two different conditions; a control condition and a TGFB-absent condition.</p>
<p>First, install <code>CiteFuse</code> using <code>BiocManager</code>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
 <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"CiteFuse"</span><span class="op">)</span></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">CiteFuse</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/">scater</a></span><span class="op">)</span>
<span class="co">#&gt; Loading required package: SingleCellExperiment</span>
<span class="co">#&gt; Loading required package: SummarizedExperiment</span>
<span class="co">#&gt; Loading required package: GenomicRanges</span>
<span class="co">#&gt; Loading required package: stats4</span>
<span class="co">#&gt; Loading required package: BiocGenerics</span>
<span class="co">#&gt; Loading required package: parallel</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'BiocGenerics'</span>
<span class="co">#&gt; The following objects are masked from 'package:parallel':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span>
<span class="co">#&gt;     clusterExport, clusterMap, parApply, parCapply, parLapply,</span>
<span class="co">#&gt;     parLapplyLB, parRapply, parSapply, parSapplyLB</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     IQR, mad, sd, var, xtabs</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span>
<span class="co">#&gt;     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span>
<span class="co">#&gt;     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span>
<span class="co">#&gt;     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span>
<span class="co">#&gt;     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span>
<span class="co">#&gt;     union, unique, unsplit, which, which.max, which.min</span>
<span class="co">#&gt; Loading required package: S4Vectors</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'S4Vectors'</span>
<span class="co">#&gt; The following object is masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     expand.grid</span>
<span class="co">#&gt; Loading required package: IRanges</span>
<span class="co">#&gt; Loading required package: GenomeInfoDb</span>
<span class="co">#&gt; Loading required package: Biobase</span>
<span class="co">#&gt; Welcome to Bioconductor</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     Vignettes contain introductory material; view with</span>
<span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span>
<span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span>
<span class="co">#&gt; Loading required package: DelayedArray</span>
<span class="co">#&gt; Loading required package: matrixStats</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'matrixStats'</span>
<span class="co">#&gt; The following objects are masked from 'package:Biobase':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     anyMissing, rowMedians</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'DelayedArray'</span>
<span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     aperm, apply, rowsum</span>
<span class="co">#&gt; Loading required package: ggplot2</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rstudio/DT">DT</a></span><span class="op">)</span></pre></div>
<p>We have made the pre-processed, integrated dataset available as a <code>SingleCellExperiment</code> object in the workshop package, which we import below.</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"CITEseq_example"</span>, package <span class="op">=</span> <span class="st">"CiteFuse"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">CITEseq_example</span><span class="op">)</span>
<span class="co">#&gt; [1] "RNA" "ADT" "HTO"</span>
<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">CITEseq_example</span>, <span class="va">dim</span><span class="op">)</span>
<span class="co">#&gt; $RNA</span>
<span class="co">#&gt; [1] 19521   500</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $ADT</span>
<span class="co">#&gt; [1]  49 500</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $HTO</span>
<span class="co">#&gt; [1]   4 500</span></pre></div>
<p>Here, we start from a list of three matrices of unique molecular identifier (UMI), antibody derived tags (ADT) and hashtag oligonucleotide (HTO) count, which have common cell names. There are 500 cells in our subsetted dataset. And characteristically of CITE-seq data, the matrices are matched, meaning that for any given cell we know the expression level of their RNA transcripts (genome-wide) and its corresponding cell surface protein expression. The <code>preprocessing</code> function will utilise the three matrices and its common cell names to create a <code>SingleCellExperiment</code> object, which stores RNA data in an <code>assay</code> and <code>ADT</code> and <code>HTO</code> data within in the <code>altExp</code> slot.</p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/preprocessing.html">preprocessing</a></span><span class="op">(</span><span class="va">CITEseq_example</span><span class="op">)</span>
<span class="va">sce_citeseq</span>
<span class="co">#&gt; class: SingleCellExperiment </span>
<span class="co">#&gt; dim: 19521 500 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): counts</span>
<span class="co">#&gt; rownames(19521): hg19_AL627309.1 hg19_AL669831.5 ... hg19_MT-ND6</span>
<span class="co">#&gt;   hg19_MT-CYB</span>
<span class="co">#&gt; rowData names(0):</span>
<span class="co">#&gt; colnames(500): AAGCCGCGTTGTCTTT GATCGCGGTTATCGGT ... TTGGCAACACTAGTAC</span>
<span class="co">#&gt;   GCTGCGAGTTGTGGCC</span>
<span class="co">#&gt; colData names(0):</span>
<span class="co">#&gt; reducedDimNames(0):</span>
<span class="co">#&gt; altExpNames(2): ADT HTO</span></pre></div>
</div>
<div id="doublet-detection-using-citefuse" class="section level2">
<h2 class="hasAnchor">
<a href="#doublet-detection-using-citefuse" class="anchor"></a>Doublet detection using <code>CiteFuse</code>
</h2>
<div id="hto-normalisation-and-visualisation" class="section level3">
<h3 class="hasAnchor">
<a href="#hto-normalisation-and-visualisation" class="anchor"></a>HTO Normalisation and Visualisation</h3>
<p>The function <code>normaliseExprs</code> is used to scale the alternative expression. Here, we used it to perform log-transformation of the <code>HTO</code> count, by setting <code>transform = "log"</code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/normaliseExprs.html">normaliseExprs</a></span><span class="op">(</span>sce <span class="op">=</span> <span class="va">sce_citeseq</span>, 
                              altExp_name <span class="op">=</span> <span class="st">"HTO"</span>, 
                              transform <span class="op">=</span> <span class="st">"log"</span><span class="op">)</span></pre></div>
<p>Then we can perform dimension reduction on the <code>HTO</code> count by using <code>runTSNE</code> or <code>runUMAP</code>, then use <code>visualiseDim</code> function to visualise the reduced dimension plot. Our CITE-seq dataset contain data from four samples that were pooled before sequencing. The samples were multiplexed through cell hashing (Stoekius et al., 2018). The four clusters observed on reduced dimension plots equate to the four different samples.</p>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/runTSNE.html">runTSNE</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, 
                               altexp <span class="op">=</span> <span class="st">"HTO"</span>, 
                               name <span class="op">=</span> <span class="st">"TSNE_HTO"</span>, 
                               pca <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visualiseDim.html">visualiseDim</a></span><span class="op">(</span><span class="va">sce_citeseq</span>,
             dimNames <span class="op">=</span> <span class="st">"TSNE_HTO"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"tSNE (HTO)"</span><span class="op">)</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-7-1.png" width="576"></p>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/runUMAP.html">runUMAP</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, 
                               altexp <span class="op">=</span> <span class="st">"HTO"</span>, 
                               name <span class="op">=</span> <span class="st">"UMAP_HTO"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visualiseDim.html">visualiseDim</a></span><span class="op">(</span><span class="va">sce_citeseq</span>,
             dimNames <span class="op">=</span> <span class="st">"UMAP_HTO"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"UMAP (HTO)"</span><span class="op">)</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-7-2.png" width="576"></p>
</div>
<div id="doublet-identification-step-1-cross-sample-doublet-detection" class="section level3">
<h3 class="hasAnchor">
<a href="#doublet-identification-step-1-cross-sample-doublet-detection" class="anchor"></a>Doublet identification step 1: cross-sample doublet detection</h3>
<p>An important step in single cell data analysis is the removal of doublets. Doublets form as a result of co-encapsulation of cells within a droplet, leading to a hybrid transcriptome from two or more cells. In CiteFuse, we implement a step-wise doublet detection approach to remove doublets. We first identify the cross-sample doublets via the <code>crossSampleDoublets</code> function.</p>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/crossSampleDoublets.html">crossSampleDoublets</a></span><span class="op">(</span><span class="va">sce_citeseq</span><span class="op">)</span>
<span class="co">#&gt; number of iterations= 20 </span>
<span class="co">#&gt; number of iterations= 24 </span>
<span class="co">#&gt; number of iterations= 46 </span>
<span class="co">#&gt; number of iterations= 50</span></pre></div>
<p>The results of the cross sample doublets are then saved in <code>colData</code> as <code>doubletClassify_between_label</code> and <code>doubletClassify_between_class</code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">sce_citeseq</span><span class="op">$</span><span class="va">doubletClassify_between_label</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;                 1                 2                 3                 4 </span>
<span class="co">#&gt;               115               121                92               129 </span>
<span class="co">#&gt; doublet/multiplet </span>
<span class="co">#&gt;                43</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">sce_citeseq</span><span class="op">$</span><span class="va">doubletClassify_between_class</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; doublet/multiplet           Singlet </span>
<span class="co">#&gt;                43               457</span></pre></div>
<p>We can then highlight the cross-sample doublets in our tSNE plot of HTO count.</p>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visualiseDim.html">visualiseDim</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, 
             dimNames <span class="op">=</span> <span class="st">"TSNE_HTO"</span>, 
             colour_by <span class="op">=</span> <span class="st">"doubletClassify_between_label"</span><span class="op">)</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-10-1.png" width="576"></p>
<p>Furthermore, <code>plotHTO</code> function allows us to plot the pairwise scatter HTO count. Any cells that show co-expression of orthologocal HTOs (red) are considered as doublets.</p>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/plotHTO.html">plotHTO</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-11-1.png" width="576"></p>
</div>
<div id="doublet-identification-step-1-within-sample-doublet-detection" class="section level3">
<h3 class="hasAnchor">
<a href="#doublet-identification-step-1-within-sample-doublet-detection" class="anchor"></a>Doublet identification step 1: within-sample doublet detection</h3>
<p>We then identify the within-sample doublets via the <code>withinSampleDoublets</code> function.</p>
<div class="sourceCode" id="cb12"><pre class="downlit">
<span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/withinSampleDoublets.html">withinSampleDoublets</a></span><span class="op">(</span><span class="va">sce_citeseq</span>,
                                    minPts <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></pre></div>
<p>The results of the cross sample doublets are then saved in the <code>colData</code> as <code>doubletClassify_within_label</code> and <code>doubletClassify_within_class</code>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">sce_citeseq</span><span class="op">$</span><span class="va">doubletClassify_within_label</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  Doublets(Within)_1  Doublets(Within)_2  Doublets(Within)_3  Doublets(Within)_4 </span>
<span class="co">#&gt;                   3                   7                   4                   6 </span>
<span class="co">#&gt; NotDoublets(Within) </span>
<span class="co">#&gt;                 480</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">sce_citeseq</span><span class="op">$</span><span class="va">doubletClassify_within_class</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Doublet Singlet </span>
<span class="co">#&gt;      20     480</span></pre></div>
<p>Again, we can visualise the within-sample doublets in our tSNE plot.</p>
<div class="sourceCode" id="cb14"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visualiseDim.html">visualiseDim</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, 
             dimNames <span class="op">=</span> <span class="st">"TSNE_HTO"</span>, 
             colour_by <span class="op">=</span> <span class="st">"doubletClassify_within_label"</span><span class="op">)</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-14-1.png" width="576"></p>
<p>Finally, we can filter out the doublet cells (both within and between batches) for the downstream analysis.</p>
<div class="sourceCode" id="cb15"><pre class="downlit">
<span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="va">sce_citeseq</span><span class="op">[</span>, <span class="va">sce_citeseq</span><span class="op">$</span><span class="va">doubletClassify_within_class</span> <span class="op">==</span> <span class="st">"Singlet"</span> <span class="op">&amp;</span> <span class="va">sce_citeseq</span><span class="op">$</span><span class="va">doubletClassify_between_class</span> <span class="op">==</span> <span class="st">"Singlet"</span><span class="op">]</span>
<span class="va">sce_citeseq</span>
<span class="co">#&gt; class: SingleCellExperiment </span>
<span class="co">#&gt; dim: 19521 437 </span>
<span class="co">#&gt; metadata(3): doubletClassify_between_threshold</span>
<span class="co">#&gt;   doubletClassify_between_resultsMat doubletClassify_within_resultsMat</span>
<span class="co">#&gt; assays(1): counts</span>
<span class="co">#&gt; rownames(19521): hg19_AL627309.1 hg19_AL669831.5 ... hg19_MT-ND6</span>
<span class="co">#&gt;   hg19_MT-CYB</span>
<span class="co">#&gt; rowData names(0):</span>
<span class="co">#&gt; colnames(437): GATCGCGGTTATCGGT GGCTGGTAGAGGTTAT ... TTGGCAACACTAGTAC</span>
<span class="co">#&gt;   GCTGCGAGTTGTGGCC</span>
<span class="co">#&gt; colData names(5): doubletClassify_between_label</span>
<span class="co">#&gt;   doubletClassify_between_class nUMI doubletClassify_within_label</span>
<span class="co">#&gt;   doubletClassify_within_class</span>
<span class="co">#&gt; reducedDimNames(2): TSNE_HTO UMAP_HTO</span>
<span class="co">#&gt; altExpNames(2): ADT HTO</span></pre></div>
</div>
</div>
<div id="clustering" class="section level2">
<h2 class="hasAnchor">
<a href="#clustering" class="anchor"></a>Clustering</h2>
<div id="performing-snf" class="section level3">
<h3 class="hasAnchor">
<a href="#performing-snf" class="anchor"></a>Performing SNF</h3>
<p>The first step of analysis is to integrate the RNA and ADT matrix. We use a popular integration algorithm called similarity network fusion (SNF) to integrate the multiomic data.</p>
<div class="sourceCode" id="cb16"><pre class="downlit">
<span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/logNormCounts.html">logNormCounts</a></span><span class="op">(</span><span class="va">sce_citeseq</span><span class="op">)</span>
<span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/normaliseExprs.html">normaliseExprs</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, altExp_name <span class="op">=</span> <span class="st">"ADT"</span>, transform <span class="op">=</span> <span class="st">"log"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span><span class="op">(</span><span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/CiteFuse.html">CiteFuse</a></span><span class="op">(</span><span class="va">sce_citeseq</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Calculating affinity matrix </span>
<span class="co">#&gt; Performing SNF</span>
<span class="co">#&gt;    user  system elapsed </span>
<span class="co">#&gt;   7.099   0.313   7.954</span></pre></div>
<p>We now proceed with the fused matrix, which is stored as <code>SNF_W</code> in our <code>sce_citeseq</code> object.</p>
</div>
<div id="performing-spectral-clustering" class="section level3">
<h3 class="hasAnchor">
<a href="#performing-spectral-clustering" class="anchor"></a>Performing spectral clustering</h3>
<p>CiteFuse implements two different clustering algorithms on the fused matrix, spectral clustering and Louvain clustering. First, we perform spectral clustering with sufficient numbers of <code>K</code> and use the eigen values to determine the optimal number of clusters.</p>
<div class="sourceCode" id="cb17"><pre class="downlit">
<span class="va">SNF_W_clust</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/spectralClustering.html">spectralClustering</a></span><span class="op">(</span><span class="fu">metadata</span><span class="op">(</span><span class="va">sce_citeseq</span><span class="op">)</span><span class="op">[[</span><span class="st">"SNF_W"</span><span class="op">]</span><span class="op">]</span>, K <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>
<span class="co">#&gt; Computing Spectral Clustering </span>
<span class="co">#&gt; Computing Diffusion Coordinates</span>
<span class="co">#&gt; Used default value: 20 dimensions</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">SNF_W_clust</span><span class="op">$</span><span class="va">eigen_values</span><span class="op">)</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-17-1.png" width="700"></p>
<div class="sourceCode" id="cb18"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/which.min.html">which.max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/diff.html">diff</a></span><span class="op">(</span><span class="va">SNF_W_clust</span><span class="op">$</span><span class="va">eigen_values</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 5</span></pre></div>
<p>Using the optimal cluster number defined from the previous step, we can now use the <code>spectralClutering</code> function to cluster the single cells by specifying the number of clusters in <code>K</code>. The function takes a cell-to-cell similarity matrix as an input. We have already created the fused similarity matrix from <code>CiteFuse</code>. Since the <code>CiteFuse</code> function creates and stores the similarity matries from ADT and RNA expression, as well the fused matrix, we can use these two to compare the clustering outcomes by data modality.</p>
<div class="sourceCode" id="cb19"><pre class="downlit">
<span class="va">SNF_W_clust</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/spectralClustering.html">spectralClustering</a></span><span class="op">(</span><span class="fu">metadata</span><span class="op">(</span><span class="va">sce_citeseq</span><span class="op">)</span><span class="op">[[</span><span class="st">"SNF_W"</span><span class="op">]</span><span class="op">]</span>, K <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>
<span class="co">#&gt; Computing Spectral Clustering </span>
<span class="co">#&gt; Computing Diffusion Coordinates</span>
<span class="co">#&gt; Used default value: 5 dimensions</span>
<span class="va">sce_citeseq</span><span class="op">$</span><span class="va">SNF_W_clust</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">SNF_W_clust</span><span class="op">$</span><span class="va">labels</span><span class="op">)</span>
<span class="va">SNF_W1_clust</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/spectralClustering.html">spectralClustering</a></span><span class="op">(</span><span class="fu">metadata</span><span class="op">(</span><span class="va">sce_citeseq</span><span class="op">)</span><span class="op">[[</span><span class="st">"ADT_W"</span><span class="op">]</span><span class="op">]</span>, K <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>
<span class="co">#&gt; Computing Spectral Clustering </span>
<span class="co">#&gt; Computing Diffusion Coordinates</span>
<span class="co">#&gt; Used default value: 5 dimensions</span>
<span class="va">sce_citeseq</span><span class="op">$</span><span class="va">ADT_clust</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">SNF_W1_clust</span><span class="op">$</span><span class="va">labels</span><span class="op">)</span>
<span class="va">SNF_W2_clust</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/spectralClustering.html">spectralClustering</a></span><span class="op">(</span><span class="fu">metadata</span><span class="op">(</span><span class="va">sce_citeseq</span><span class="op">)</span><span class="op">[[</span><span class="st">"RNA_W"</span><span class="op">]</span><span class="op">]</span>, K <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>
<span class="co">#&gt; Computing Spectral Clustering </span>
<span class="co">#&gt; Computing Diffusion Coordinates</span>
<span class="co">#&gt; Used default value: 5 dimensions</span>
<span class="va">sce_citeseq</span><span class="op">$</span><span class="va">RNA_clust</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">SNF_W2_clust</span><span class="op">$</span><span class="va">labels</span><span class="op">)</span></pre></div>
</div>
<div id="visualisation" class="section level3">
<h3 class="hasAnchor">
<a href="#visualisation" class="anchor"></a>Visualisation</h3>
<p>The outcome of the clustering can be easily visualised on a reduced dimensions plot by highlighting the points by cluster label.</p>
<div class="sourceCode" id="cb20"><pre class="downlit">
<span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/reducedDimSNF.html">reducedDimSNF</a></span><span class="op">(</span><span class="va">sce_citeseq</span>,
                             method <span class="op">=</span> <span class="st">"tSNE"</span>, 
                             dimNames <span class="op">=</span> <span class="st">"tSNE_joint"</span><span class="op">)</span>
<span class="va">g1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visualiseDim.html">visualiseDim</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, dimNames <span class="op">=</span> <span class="st">"tSNE_joint"</span>, colour_by <span class="op">=</span> <span class="st">"SNF_W_clust"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"tSNE (SNF clustering)"</span><span class="op">)</span>
<span class="va">g2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visualiseDim.html">visualiseDim</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, dimNames <span class="op">=</span> <span class="st">"tSNE_joint"</span>,  colour_by <span class="op">=</span> <span class="st">"ADT_clust"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"tSNE (ADT clustering)"</span><span class="op">)</span>
<span class="va">g3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visualiseDim.html">visualiseDim</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, dimNames <span class="op">=</span> <span class="st">"tSNE_joint"</span>,  colour_by <span class="op">=</span> <span class="st">"RNA_clust"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"tSNE (RNA clustering)"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">gridExtra</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'gridExtra'</span>
<span class="co">#&gt; The following object is masked from 'package:Biobase':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     combine</span>
<span class="co">#&gt; The following object is masked from 'package:BiocGenerics':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     combine</span>
<span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html">grid.arrange</a></span><span class="op">(</span><span class="va">g3</span>, <span class="va">g2</span>, <span class="va">g1</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-19-1.png" width="768"></p>
<p>The expression of genes and proteins can be visualised by changing the <code>colour_by</code> parameter to assess the clusters. As an example, we highlight the plot by the RNA and ADT expression level of CD8.</p>
<div class="sourceCode" id="cb21"><pre class="downlit">
<span class="va">g1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visualiseDim.html">visualiseDim</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, dimNames <span class="op">=</span> <span class="st">"tSNE_joint"</span>, 
                   colour_by <span class="op">=</span> <span class="st">"hg19_CD8A"</span>,
                   data_from <span class="op">=</span> <span class="st">"assay"</span>,
                   assay_name <span class="op">=</span> <span class="st">"logcounts"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"tSNE: hg19_CD8A (RNA expression)"</span><span class="op">)</span>
<span class="va">g2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visualiseDim.html">visualiseDim</a></span><span class="op">(</span><span class="va">sce_citeseq</span>,dimNames <span class="op">=</span> <span class="st">"tSNE_joint"</span>, 
                   colour_by <span class="op">=</span> <span class="st">"CD8"</span>,
                   data_from <span class="op">=</span> <span class="st">"altExp"</span>,
                   altExp_assay_name <span class="op">=</span> <span class="st">"logcounts"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"tSNE: CD8 (ADT expression)"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html">grid.arrange</a></span><span class="op">(</span><span class="va">g1</span>, <span class="va">g2</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-20-1.png" width="700"></p>
</div>
<div id="louvain-clustering" class="section level3">
<h3 class="hasAnchor">
<a href="#louvain-clustering" class="anchor"></a>Louvain clustering</h3>
<p>As well as spectral clustering, CiteFuse can implement Louvain clustering if users wish to use another clustering method. We use the <code>igraph</code> package, and any community detection algorithms available in their package can be selected by changing the <code>method</code> parameter.</p>
<div class="sourceCode" id="cb22"><pre class="downlit">
<span class="va">SNF_W_louvain</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/igraphClustering.html">igraphClustering</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, method <span class="op">=</span> <span class="st">"louvain"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">SNF_W_louvain</span><span class="op">)</span>
<span class="co">#&gt; SNF_W_louvain</span>
<span class="co">#&gt;  1  2  3  4  5  6  7  8  9 </span>
<span class="co">#&gt; 30 38 80 29 90 37 52 29 52</span>
<span class="va">sce_citeseq</span><span class="op">$</span><span class="va">SNF_W_louvain</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">SNF_W_louvain</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visualiseDim.html">visualiseDim</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, dimNames <span class="op">=</span> <span class="st">"tSNE_joint"</span>, colour_by <span class="op">=</span> <span class="st">"SNF_W_louvain"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"tSNE (SNF louvain clustering)"</span><span class="op">)</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-21-1.png" width="768"></p>
<div class="sourceCode" id="cb23"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visualiseKNN.html">visualiseKNN</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, colour_by <span class="op">=</span> <span class="st">"SNF_W_louvain"</span><span class="op">)</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-22-1.png" width="576"></p>
</div>
</div>
<div id="differential-expression-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#differential-expression-analysis" class="anchor"></a>Differential Expression Analysis</h2>
<div id="exploration-of-feature-expression" class="section level3">
<h3 class="hasAnchor">
<a href="#exploration-of-feature-expression" class="anchor"></a>Exploration of feature expression</h3>
<p>CiteFuse has a wide range of visualisation tools to facilitate exploratory analysis of CITE-seq data. The <code>visualiseExprs</code> function is an easy-to-use function to generate boxplots, violinplots, jitter plots, density plots, and pairwise scatter/density plots of genes and proteins expressed in the data. The plots can be grouped by using the cluster labels stored in the <code>sce_citeseq</code> object.</p>
<div class="sourceCode" id="cb24"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visualiseExprs.html">visualiseExprs</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, 
               plot <span class="op">=</span> <span class="st">"boxplot"</span>, 
               group_by <span class="op">=</span> <span class="st">"SNF_W_louvain"</span>,
               feature_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"hg19_CD2"</span>, <span class="st">"hg19_CD4"</span>, <span class="st">"hg19_CD8A"</span>, <span class="st">"hg19_CD19"</span><span class="op">)</span><span class="op">)</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-23-1.png" width="768"></p>
<div class="sourceCode" id="cb25"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visualiseExprs.html">visualiseExprs</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, 
               plot <span class="op">=</span> <span class="st">"violin"</span>, 
               group_by <span class="op">=</span> <span class="st">"SNF_W_louvain"</span>,
               feature_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"hg19_CD2"</span>, <span class="st">"hg19_CD4"</span>, <span class="st">"hg19_CD8A"</span>, <span class="st">"hg19_CD19"</span><span class="op">)</span><span class="op">)</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-23-2.png" width="768"></p>
<div class="sourceCode" id="cb26"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visualiseExprs.html">visualiseExprs</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, 
               plot <span class="op">=</span> <span class="st">"jitter"</span>, 
               group_by <span class="op">=</span> <span class="st">"SNF_W_louvain"</span>,
               feature_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"hg19_CD2"</span>, <span class="st">"hg19_CD4"</span>, <span class="st">"hg19_CD8A"</span>, <span class="st">"hg19_CD19"</span><span class="op">)</span><span class="op">)</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-23-3.png" width="768"></p>
<div class="sourceCode" id="cb27"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visualiseExprs.html">visualiseExprs</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, 
               plot <span class="op">=</span> <span class="st">"density"</span>, 
               group_by <span class="op">=</span> <span class="st">"SNF_W_louvain"</span>,
               feature_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"hg19_CD2"</span>, <span class="st">"hg19_CD4"</span>, <span class="st">"hg19_CD8A"</span>, <span class="st">"hg19_CD19"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Picking joint bandwidth of 0.277</span>
<span class="co">#&gt; Picking joint bandwidth of 0.354</span>
<span class="co">#&gt; Picking joint bandwidth of 0.248</span>
<span class="co">#&gt; Picking joint bandwidth of 0.388</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-23-4.png" width="768"></p>
<div class="sourceCode" id="cb28"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visualiseExprs.html">visualiseExprs</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, 
               altExp_name <span class="op">=</span> <span class="st">"ADT"</span>, 
               group_by <span class="op">=</span> <span class="st">"SNF_W_louvain"</span>,
               plot <span class="op">=</span> <span class="st">"violin"</span>, n <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-24-1.png" width="576"></p>
<div class="sourceCode" id="cb29"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visualiseExprs.html">visualiseExprs</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, altExp_name <span class="op">=</span> <span class="st">"ADT"</span>, 
               plot <span class="op">=</span> <span class="st">"jitter"</span>, 
               group_by <span class="op">=</span> <span class="st">"SNF_W_louvain"</span>, 
               feature_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"CD2"</span>, <span class="st">"CD8"</span>, <span class="st">"CD4"</span>, <span class="st">"CD19"</span><span class="op">)</span><span class="op">)</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-24-2.png" width="576"></p>
<div class="sourceCode" id="cb30"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visualiseExprs.html">visualiseExprs</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, altExp_name <span class="op">=</span> <span class="st">"ADT"</span>, 
               plot <span class="op">=</span> <span class="st">"density"</span>, 
               group_by <span class="op">=</span> <span class="st">"SNF_W_louvain"</span>,
               feature_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"CD2"</span>, <span class="st">"CD8"</span>, <span class="st">"CD4"</span>, <span class="st">"CD19"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Picking joint bandwidth of 0.274</span>
<span class="co">#&gt; Picking joint bandwidth of 0.362</span>
<span class="co">#&gt; Picking joint bandwidth of 0.285</span>
<span class="co">#&gt; Picking joint bandwidth of 0.28</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-24-3.png" width="576"></p>
<div class="sourceCode" id="cb31"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visualiseExprs.html">visualiseExprs</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, altExp_name <span class="op">=</span> <span class="st">"ADT"</span>, 
               plot <span class="op">=</span> <span class="st">"pairwise"</span>, 
               feature_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"CD4"</span>, <span class="st">"CD8"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; number of iterations= 13 </span>
<span class="co">#&gt; number of iterations= 18</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-25-1.png" width="768"></p>
<div class="sourceCode" id="cb32"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visualiseExprs.html">visualiseExprs</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, altExp_name <span class="op">=</span> <span class="st">"ADT"</span>, 
               plot <span class="op">=</span> <span class="st">"pairwise"</span>, 
               feature_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"CD45RA"</span>, <span class="st">"CD4"</span>, <span class="st">"CD8"</span><span class="op">)</span>, 
               threshold <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fl">4</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-25-2.png" width="768"></p>
</div>
</div>
<div id="perform-de-analysis-with-wilcoxon-rank-sum-test" class="section level2">
<h2 class="hasAnchor">
<a href="#perform-de-analysis-with-wilcoxon-rank-sum-test" class="anchor"></a>Perform DE Analysis with Wilcoxon Rank Sum test</h2>
<p>CiteFuse also calculates differentially expressed (DE) genes through the <code>DEgenes</code> function. The cluster grouping to use must be specified in the <code>group</code> parameter. If <code>altExp_name</code> is not specified, RNA expression will be used as the default expression matrix.</p>
<p>Results form the DE analysis is stored in <code>sce_citeseq</code> as <code>DE_res_RNA_filter</code> and <code>DE_res_ADT_filter</code> for RNA and ADT expression, respectively.</p>
<div id="for-rna-expression" class="section level3">
<h3 class="hasAnchor">
<a href="#for-rna-expression" class="anchor"></a>For RNA expression</h3>
<div class="sourceCode" id="cb33"><pre class="downlit">
<span class="co"># DE will be performed for RNA if altExp_name = "none" </span>
<span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/DEgenes.html">DEgenes</a></span><span class="op">(</span><span class="va">sce_citeseq</span>,
                       altExp_name <span class="op">=</span> <span class="st">"none"</span>, 
                       group <span class="op">=</span> <span class="va">sce_citeseq</span><span class="op">$</span><span class="va">SNF_W_louvain</span>,
                       return_all <span class="op">=</span> <span class="cn">TRUE</span>,
                       exprs_pct <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>
<span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/selectDEgenes.html">selectDEgenes</a></span><span class="op">(</span><span class="va">sce_citeseq</span>,
                             altExp_name <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html">datatable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/format.html">format</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu">metadata</span><span class="op">(</span><span class="va">sce_citeseq</span><span class="op">)</span><span class="op">[[</span><span class="st">"DE_res_RNA_filter"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>, 
                 digits <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span></pre></div>
<div id="htmlwidget-9f0426b98fd78476bbe0" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-9f0426b98fd78476bbe0">{"x":{"filter":"none","data":[["1.hg19_GZMK","1.hg19_KLRB1","1.hg19_DUSP2","1.hg19_IL7R","1.hg19_NCR3","1.hg19_CEBPD","1.hg19_KLRG1","1.hg19_CD69","1.hg19_LINC01871","1.hg19_IL32","2.hg19_CD8B","2.hg19_CMC1","2.hg19_CCL5","2.hg19_CD8A","2.hg19_GZMM","2.hg19_DUSP2","2.hg19_NDUFA6","2.hg19_HCST","2.hg19_RNF166","2.hg19_FYN","3.hg19_ZNF683","3.hg19_TRBV20-1","3.hg19_CD8A","3.hg19_FGFBP2","3.hg19_GZMH","3.hg19_CD8B","3.hg19_CCL5","3.hg19_CST7","3.hg19_KLRD1","3.hg19_CYTOR","4.hg19_CCR7","4.hg19_ACTN1","4.hg19_NELL2","4.hg19_PASK","4.hg19_MAL","4.hg19_CD8B","4.hg19_LEF1","4.hg19_PIK3IP1","4.hg19_LDLRAP1","4.hg19_PCED1B","5.hg19_LTB","5.hg19_TCF7","5.hg19_IL7R","5.hg19_LDHB","5.hg19_INPP4B","5.hg19_CD27","5.hg19_LEF1","5.hg19_NOSIP","5.hg19_SELL","5.hg19_CAMK4","6.hg19_FCN1","6.hg19_LYZ","6.hg19_SPI1","6.hg19_CYBB","6.hg19_HLA-DRA","6.hg19_S100A9","6.hg19_CD68","6.hg19_S100A8","6.hg19_CSTA","6.hg19_VCAN","7.hg19_GNLY","7.hg19_CCL5","7.hg19_GZMB","7.hg19_CTSW","7.hg19_NKG7","7.hg19_GZMH","7.hg19_IFITM2","7.hg19_KLRD1","7.hg19_CD8A","7.hg19_TYROBP","8.hg19_IL7R","8.hg19_S100A4","8.hg19_S100A10","8.hg19_IL32","8.hg19_EML4","8.hg19_FAM107B","8.hg19_RORA","8.hg19_CD2","8.hg19_TNFAIP3","8.hg19_GUK1","9.hg19_SH2D1B","9.hg19_KLRF1","9.hg19_CLIC3","9.hg19_FCER1G","9.hg19_MYOM2","9.hg19_TRDC","9.hg19_SPON2","9.hg19_TYROBP","9.hg19_IL2RB","9.hg19_KLRB1"],["2668"," 850","2126","2159","3396","3393","2878","3109","3983","3478","3257","3984","3217","4004","4452","5063","5205","4890","5330","5168","4890","6321","4268","5222","5277","7670","6396","6720","7320","8196","2449","2815","2907","3163","3115","2175","2384","2787","3260","3524","3917","5592","5443","5736","9061","9080","8530","7904","8680","9426","2400","2406","2409","2600"," 661","2808","3200","2825","3400","3400","2545","3656","4591","5072","4800","5220","4839","5452","5463","6141","2602","3100","3253","3434","3973","4176","4256","4141","4160","4118","3603","2742","3449","3498","2047","4272","2804","2748","3121","3143"],["7.8e-24","2.5e-23","1.1e-11","1.2e-10","1.6e-10","3.2e-10","9.2e-09","2.5e-07","7.5e-06","7.6e-05","8.8e-14","9.8e-10","2.0e-09","1.6e-07","4.5e-06","1.1e-04","1.7e-04","2.7e-04","2.9e-04","3.3e-04","9.3e-43","8.1e-37","1.2e-26","9.8e-23","8.4e-22","9.8e-17","2.9e-15","2.9e-14","7.9e-14","1.5e-13","3.0e-21","3.7e-19","8.1e-19","1.6e-18","1.4e-13","2.8e-13","5.0e-13","5.0e-11","1.5e-10","2.7e-10","3.7e-34","2.4e-31","3.1e-25","7.1e-22","2.0e-19","3.8e-19","4.7e-19","2.3e-16","2.9e-15","1.0e-14","4.5e-64","1.1e-61","1.3e-61","1.9e-61","9.1e-60","2.2e-56","1.3e-53","4.7e-52","4.8e-51","4.8e-51","3.1e-19","2.9e-14","7.3e-12","1.9e-10","4.8e-10","1.1e-09","1.4e-09","5.1e-09","6.9e-09","7.8e-09","3.9e-08","1.8e-05","4.5e-05","1.5e-04","4.2e-04","2.8e-03","3.2e-03","3.9e-03","4.1e-03","4.6e-03","7.0e-52","3.7e-37","4.7e-34","1.4e-33","1.6e-33","6.8e-31","1.4e-27","2.3e-27","8.4e-26","3.1e-24"],["7.8e-24","2.5e-23","1.1e-11","1.2e-10","1.6e-10","3.2e-10","9.2e-09","2.5e-07","7.5e-06","7.6e-05","8.8e-14","9.8e-10","2.0e-09","1.6e-07","4.5e-06","1.1e-04","1.7e-04","2.7e-04","2.9e-04","3.3e-04","9.3e-43","8.1e-37","1.2e-26","9.8e-23","8.4e-22","9.8e-17","2.9e-15","2.9e-14","7.9e-14","1.5e-13","3.0e-21","3.7e-19","8.1e-19","1.6e-18","1.4e-13","2.8e-13","5.0e-13","5.0e-11","1.5e-10","2.7e-10","3.7e-34","2.4e-31","3.1e-25","7.1e-22","2.0e-19","3.8e-19","4.7e-19","2.3e-16","2.9e-15","1.0e-14","4.5e-64","1.1e-61","1.3e-61","1.9e-61","9.1e-60","2.2e-56","1.3e-53","4.7e-52","4.8e-51","4.8e-51","3.1e-19","2.9e-14","7.3e-12","1.9e-10","4.8e-10","1.1e-09","1.4e-09","5.1e-09","6.9e-09","7.8e-09","3.9e-08","1.8e-05","4.5e-05","1.5e-04","4.2e-04","2.8e-03","3.2e-03","3.9e-03","4.1e-03","4.6e-03","7.0e-52","3.7e-37","4.7e-34","1.4e-33","1.6e-33","6.8e-31","1.4e-27","2.3e-27","8.4e-26","3.1e-24"],["0.1088","0.4686","0.5104","0.9285","0.1657","0.2075","0.4686","0.4013","0.2302","1.7839","0.3541","0.4049","2.0103","0.8258","0.5960","0.5444","0.3662","1.3533","0.3648","0.5329","0.0920","0.1043","0.6408","0.6099","0.7164","0.3141","1.8257","1.3502","0.6572","0.2635","0.0890","0.0690","0.0740","0.0513","0.1080","0.3614","0.2537","0.2096","0.1353","0.0991","0.5608","0.2284","0.7299","0.7113","0.1061","0.1245","0.1797","0.3680","0.3325","0.1754","0.0000","0.0016","0.0027","0.0000","0.0529","0.0020","0.0000","0.0079","0.0000","0.0000","2.3480","1.9339","0.9406","0.6517","2.1203","0.8427","1.6395","0.7225","0.7999","0.5668","0.9520","2.3277","1.2842","1.7878","0.3654","0.4598","0.4339","0.6678","0.6590","0.7686","0.0184","0.1293","0.0921","0.1812","0.2495","0.0941","0.2766","0.4647","0.2215","0.4411"],["0.92","2.80","1.69","2.53","0.81","0.94","1.28","1.07","0.69","2.67","1.36","1.44","3.78","1.75","1.23","1.09","0.77","1.96","0.77","0.99","1.15","1.49","2.09","1.93","2.07","1.01","3.67","2.64","1.56","0.92","0.82","0.65","0.56","0.51","0.60","1.57","1.09","0.86","0.66","0.46","2.36","1.19","2.23","1.79","0.63","0.60","0.81","1.06","1.03","0.64","1.83","2.74","1.14","0.81","2.84","2.53","0.74","2.81","1.04","1.50","5.13","3.87","2.19","1.52","3.88","1.87","2.54","1.56","1.69","1.36","2.26","3.32","2.00","2.65","0.85","0.80","0.82","1.07","1.17","1.14","0.91","1.22","1.42","1.71","1.81","0.96","1.78","2.11","1.34","2.02"],["0.0565","0.2285","0.3489","0.4251","0.1278","0.1351","0.2998","0.3464","0.1843","0.7617","0.2206","0.2431","0.6140","0.4185","0.4286","0.3609","0.3283","0.7569","0.3058","0.4110","0.0616","0.0448","0.3529","0.3305","0.3585","0.1793","0.5686","0.5490","0.3641","0.2297","0.0784","0.0686","0.0637","0.0515","0.0931","0.2304","0.1985","0.1838","0.1275","0.0980","0.2911","0.1671","0.3314","0.5072","0.1037","0.1037","0.1441","0.3055","0.2248","0.1585","0.0000","0.0025","0.0025","0.0000","0.0450","0.0025","0.0000","0.0075","0.0000","0.0000","0.5792","0.6000","0.4156","0.3844","0.6052","0.4078","0.8312","0.3974","0.4078","0.2208","0.4265","0.8946","0.7451","0.7672","0.3137","0.3824","0.3382","0.4853","0.4755","0.5784","0.0130","0.0935","0.0831","0.0779","0.1766","0.0597","0.1896","0.1818","0.1844","0.2000"],["0.63","0.97","0.90","0.93","0.57","0.57","0.83","0.77","0.53","1.00","0.76","0.66","1.00","0.89","0.76","0.66","0.55","0.92","0.55","0.66","0.71","0.60","0.94","0.89","0.93","0.66","1.00","0.99","0.82","0.59","0.66","0.59","0.59","0.52","0.59","0.79","0.76","0.72","0.55","0.52","0.93","0.79","0.96","0.87","0.51","0.52","0.59","0.72","0.66","0.56","0.68","0.68","0.68","0.65","0.92","0.62","0.57","0.62","0.54","0.54","1.00","1.00","0.92","0.83","1.00","0.87","0.98","0.83","0.85","0.65","0.93","1.00","0.97","0.93","0.59","0.62","0.59","0.72","0.69","0.79","0.65","0.81","0.69","0.73","0.88","0.63","0.81","0.94","0.79","0.87"],["0.58","0.74","0.55","0.51","0.44","0.43","0.53","0.42","0.35","0.24","0.54","0.41","0.39","0.48","0.33","0.30","0.22","0.16","0.25","0.25","0.65","0.56","0.58","0.56","0.57","0.48","0.43","0.44","0.46","0.36","0.58","0.52","0.52","0.47","0.49","0.56","0.56","0.54","0.42","0.42","0.64","0.62","0.62","0.36","0.41","0.42","0.44","0.42","0.43","0.40","0.68","0.67","0.67","0.65","0.87","0.62","0.57","0.61","0.54","0.54","0.42","0.40","0.51","0.44","0.39","0.46","0.15","0.43","0.44","0.43","0.50","0.11","0.22","0.16","0.27","0.24","0.25","0.24","0.21","0.21","0.64","0.71","0.61","0.65","0.71","0.57","0.62","0.76","0.60","0.67"],["0.58","0.74","0.55","0.51","0.44","0.43","0.53","0.42","0.35","0.24","0.54","0.41","0.39","0.48","0.33","0.30","0.22","0.16","0.25","0.25","0.65","0.56","0.58","0.56","0.57","0.48","0.43","0.44","0.46","0.36","0.58","0.52","0.52","0.47","0.49","0.56","0.56","0.54","0.42","0.42","0.64","0.62","0.62","0.36","0.41","0.42","0.44","0.42","0.43","0.40","0.68","0.67","0.67","0.65","0.87","0.62","0.57","0.61","0.54","0.54","0.42","0.40","0.51","0.44","0.39","0.46","0.15","0.43","0.44","0.43","0.50","0.11","0.22","0.16","0.27","0.24","0.25","0.24","0.21","0.21","0.64","0.71","0.61","0.65","0.71","0.57","0.62","0.76","0.60","0.67"],["hg19_GZMK","hg19_KLRB1","hg19_DUSP2","hg19_IL7R","hg19_NCR3","hg19_CEBPD","hg19_KLRG1","hg19_CD69","hg19_LINC01871","hg19_IL32","hg19_CD8B","hg19_CMC1","hg19_CCL5","hg19_CD8A","hg19_GZMM","hg19_DUSP2","hg19_NDUFA6","hg19_HCST","hg19_RNF166","hg19_FYN","hg19_ZNF683","hg19_TRBV20-1","hg19_CD8A","hg19_FGFBP2","hg19_GZMH","hg19_CD8B","hg19_CCL5","hg19_CST7","hg19_KLRD1","hg19_CYTOR","hg19_CCR7","hg19_ACTN1","hg19_NELL2","hg19_PASK","hg19_MAL","hg19_CD8B","hg19_LEF1","hg19_PIK3IP1","hg19_LDLRAP1","hg19_PCED1B","hg19_LTB","hg19_TCF7","hg19_IL7R","hg19_LDHB","hg19_INPP4B","hg19_CD27","hg19_LEF1","hg19_NOSIP","hg19_SELL","hg19_CAMK4","hg19_FCN1","hg19_LYZ","hg19_SPI1","hg19_CYBB","hg19_HLA-DRA","hg19_S100A9","hg19_CD68","hg19_S100A8","hg19_CSTA","hg19_VCAN","hg19_GNLY","hg19_CCL5","hg19_GZMB","hg19_CTSW","hg19_NKG7","hg19_GZMH","hg19_IFITM2","hg19_KLRD1","hg19_CD8A","hg19_TYROBP","hg19_IL7R","hg19_S100A4","hg19_S100A10","hg19_IL32","hg19_EML4","hg19_FAM107B","hg19_RORA","hg19_CD2","hg19_TNFAIP3","hg19_GUK1","hg19_SH2D1B","hg19_KLRF1","hg19_CLIC3","hg19_FCER1G","hg19_MYOM2","hg19_TRDC","hg19_SPON2","hg19_TYROBP","hg19_IL2RB","hg19_KLRB1"],["1","1","1","1","1","1","1","1","1","1","2","2","2","2","2","2","2","2","2","2","3","3","3","3","3","3","3","3","3","3","4","4","4","4","4","4","4","4","4","4","5","5","5","5","5","5","5","5","5","5","6","6","6","6","6","6","6","6","6","6","7","7","7","7","7","7","7","7","7","7","8","8","8","8","8","8","8","8","8","8","9","9","9","9","9","9","9","9","9","9"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>stats.W<\/th>\n      <th>pval<\/th>\n      <th>p.adjust<\/th>\n      <th>meanExprs.1<\/th>\n      <th>meanExprs.2<\/th>\n      <th>meanPct.1<\/th>\n      <th>meanPct.2<\/th>\n      <th>meanDiff<\/th>\n      <th>pctDiff<\/th>\n      <th>name<\/th>\n      <th>group<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="for-adt-count" class="section level3">
<h3 class="hasAnchor">
<a href="#for-adt-count" class="anchor"></a>For ADT count</h3>
<div class="sourceCode" id="cb34"><pre class="downlit">
<span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/DEgenes.html">DEgenes</a></span><span class="op">(</span><span class="va">sce_citeseq</span>,
                       altExp_name <span class="op">=</span> <span class="st">"ADT"</span>, 
                       group <span class="op">=</span> <span class="va">sce_citeseq</span><span class="op">$</span><span class="va">SNF_W_louvain</span>,
                       return_all <span class="op">=</span> <span class="cn">TRUE</span>,
                       exprs_pct <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>
<span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/selectDEgenes.html">selectDEgenes</a></span><span class="op">(</span><span class="va">sce_citeseq</span>,
                             altExp_name <span class="op">=</span> <span class="st">"ADT"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html">datatable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/format.html">format</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="fu">metadata</span><span class="op">(</span><span class="va">sce_citeseq</span><span class="op">)</span><span class="op">[[</span><span class="st">"DE_res_ADT_filter"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>, 
                 digits <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span></pre></div>
<div id="htmlwidget-2751056391cefa549b18" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-2751056391cefa549b18">{"x":{"filter":"none","data":[["1.CD26 (Adenosine)","1.CD28","4","5.CD28","5.CD26 (Adenosine)","5.CD3","6.LAMP1","6.CD45RO","6.CD45","6.B220 (CD45R)","6.CD14","6.B7-H1 (PD-L1)","6.Siglec-8","6.PD-L1 (CD274)","6.TCRg","6.EpCAM (CD326)","9"],[" 2830"," 3124"," 4549"," 4114","10836","10830"," 1652"," 1660"," 2274"," 2366"," 3156"," 2859"," 2594"," 2604"," 2774"," 2860"," 3146"],["5.0e-08","7.3e-06","3.7e-02","2.6e-27","6.5e-07","6.8e-06","2.0e-15","3.0e-15","2.9e-13","2.0e-12","1.2e-11","1.7e-11","2.0e-11","4.1e-11","4.2e-11","8.1e-11","4.9e-17"],["5.0e-08","7.3e-06","3.7e-02","2.6e-27","6.5e-07","6.8e-06","2.0e-15","3.0e-15","2.9e-13","2.0e-12","1.2e-11","1.7e-11","2.0e-11","4.1e-11","4.2e-11","8.1e-11","4.9e-17"],["0.46","1.48","1.51","1.28","0.45","1.43","0.92","1.13","0.57","0.79","0.27","0.41","0.78","1.13","0.56","0.54","0.65"],["1.59","2.28","1.87","2.54","0.86","1.83","2.55","2.59","1.46","1.71","1.59","1.32","1.80","2.16","1.44","1.51","1.74"],["0.41","0.85","0.85","0.83","0.38","0.85","0.77","0.85","0.58","0.75","0.33","0.46","0.72","0.87","0.57","0.56","0.54"],["0.73","0.97","1.00","0.99","0.63","0.97","1.00","1.00","0.95","0.97","0.70","0.84","0.95","0.97","0.86","0.89","0.94"],["0.32","0.11","0.15","0.16","0.25","0.12","0.23","0.15","0.37","0.22","0.38","0.38","0.22","0.10","0.30","0.33","0.40"],["0.32","0.11","0.15","0.16","0.25","0.12","0.23","0.15","0.37","0.22","0.38","0.38","0.22","0.10","0.30","0.33","0.40"],["CD26 (Adenosine)","CD28","CD28","CD28","CD26 (Adenosine)","CD3","LAMP1","CD45RO","CD45","B220 (CD45R)","CD14","B7-H1 (PD-L1)","Siglec-8","PD-L1 (CD274)","TCRg","EpCAM (CD326)","CD366 (tim3)"],["1","1","4","5","5","5","6","6","6","6","6","6","6","6","6","6","9"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>stats.W<\/th>\n      <th>pval<\/th>\n      <th>p.adjust<\/th>\n      <th>meanExprs.1<\/th>\n      <th>meanExprs.2<\/th>\n      <th>meanPct.1<\/th>\n      <th>meanPct.2<\/th>\n      <th>meanDiff<\/th>\n      <th>pctDiff<\/th>\n      <th>name<\/th>\n      <th>group<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="visualising-de-results" class="section level2">
<h2 class="hasAnchor">
<a href="#visualising-de-results" class="anchor"></a>Visualising DE Results</h2>
<p>The DE genes can be visualised with the <code>DEbubblePlot</code> and <code>DEcomparisonPlot</code>. In each case, the gene names must first be extracted from the DE result objects.</p>
<div id="circlepackplot" class="section level3">
<h3 class="hasAnchor">
<a href="#circlepackplot" class="anchor"></a>circlepackPlot</h3>
<p>The <code>circlepackPlot</code> takes a list of all DE genes from RNA and ADT DE analysis and will plot only the top most significant DE genes to plot.</p>
<div class="sourceCode" id="cb35"><pre class="downlit">
<span class="va">rna_DEgenes</span> <span class="op">&lt;-</span> <span class="fu">metadata</span><span class="op">(</span><span class="va">sce_citeseq</span><span class="op">)</span><span class="op">[[</span><span class="st">"DE_res_RNA_filter"</span><span class="op">]</span><span class="op">]</span>
<span class="va">adt_DEgenes</span> <span class="op">&lt;-</span> <span class="fu">metadata</span><span class="op">(</span><span class="va">sce_citeseq</span><span class="op">)</span><span class="op">[[</span><span class="st">"DE_res_ADT_filter"</span><span class="op">]</span><span class="op">]</span>
<span class="va">rna_DEgenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">rna_DEgenes</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span>
  <span class="va">x</span><span class="op">$</span><span class="va">name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"hg19_"</span>, <span class="st">""</span>, <span class="va">x</span><span class="op">$</span><span class="va">name</span><span class="op">)</span>
  <span class="va">x</span><span class="op">}</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/DEbubblePlot.html">DEbubblePlot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>RNA <span class="op">=</span> <span class="va">rna_DEgenes</span>, ADT <span class="op">=</span> <span class="va">adt_DEgenes</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Non-leaf weights ignored</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-28-1.png" width="960"></p>
</div>
<div id="decomparisonplot" class="section level3">
<h3 class="hasAnchor">
<a href="#decomparisonplot" class="anchor"></a>DEcomparisonPlot</h3>
<p>For the <code>DEcomparisonPlot</code>, as well as a list containing the DE genes for RNA and ADT, a <code>feature_list</code> specifying the genes and proteins of interest is required.</p>
<div class="sourceCode" id="cb36"><pre class="downlit">
<span class="va">rna_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"hg19_CD4"</span>,
              <span class="st">"hg19_CD8A"</span>,
              <span class="st">"hg19_HLA-DRB1"</span>,
              <span class="st">"hg19_ITGAX"</span>,
              <span class="st">"hg19_NCAM1"</span>,
              <span class="st">"hg19_CD27"</span>,
              <span class="st">"hg19_CD19"</span><span class="op">)</span>
<span class="va">adt_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"CD4"</span>, <span class="st">"CD8"</span>, <span class="st">"MHCII (HLA-DR)"</span>, <span class="st">"CD11c"</span>, <span class="st">"CD56"</span>, <span class="st">"CD27"</span>, <span class="st">"CD19"</span><span class="op">)</span>
<span class="va">rna_DEgenes_all</span> <span class="op">&lt;-</span> <span class="fu">metadata</span><span class="op">(</span><span class="va">sce_citeseq</span><span class="op">)</span><span class="op">[[</span><span class="st">"DE_res_RNA"</span><span class="op">]</span><span class="op">]</span>
<span class="va">adt_DEgenes_all</span> <span class="op">&lt;-</span> <span class="fu">metadata</span><span class="op">(</span><span class="va">sce_citeseq</span><span class="op">)</span><span class="op">[[</span><span class="st">"DE_res_ADT"</span><span class="op">]</span><span class="op">]</span>
<span class="va">feature_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>RNA <span class="op">=</span> <span class="va">rna_list</span>, ADT <span class="op">=</span> <span class="va">adt_list</span><span class="op">)</span>
<span class="va">de_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>RNA <span class="op">=</span> <span class="va">rna_DEgenes_all</span>, ADT <span class="op">=</span> <span class="va">adt_DEgenes_all</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/DEcomparisonPlot.html">DEcomparisonPlot</a></span><span class="op">(</span>de_list <span class="op">=</span> <span class="va">de_list</span>,
                 feature_list <span class="op">=</span> <span class="va">feature_list</span><span class="op">)</span>
<span class="co">#&gt; Warning: Use of `df$group` is discouraged. Use `group` instead.</span>

<span class="co">#&gt; Warning: Use of `df$group` is discouraged. Use `group` instead.</span>

<span class="co">#&gt; Warning: Use of `df$group` is discouraged. Use `group` instead.</span>

<span class="co">#&gt; Warning: Use of `df$group` is discouraged. Use `group` instead.</span>

<span class="co">#&gt; Warning: Use of `df$group` is discouraged. Use `group` instead.</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-29-1.png" width="768"></p>
</div>
</div>
<div id="adt-importance-evaluation" class="section level2">
<h2 class="hasAnchor">
<a href="#adt-importance-evaluation" class="anchor"></a>ADT Importance Evaluation</h2>
<p>An important evaluation in CITE-seq data analysis is to assess the quality of each ADT and to evaluate the contribution of ADTs towards clustering outcome. CiteFuse calculates the relative importance of ADT towards clustering outcome by using a random forest model. The higher the score of an ADT, the greater its importance towards the final clustering outcome.</p>
<div class="sourceCode" id="cb37"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">2020</span><span class="op">)</span>
<span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/importanceADT.html">importanceADT</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, 
                             group <span class="op">=</span> <span class="va">sce_citeseq</span><span class="op">$</span><span class="va">SNF_W_louvain</span>,
                             subsample <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visImportance.html">visImportance</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, plot <span class="op">=</span> <span class="st">"boxplot"</span><span class="op">)</span>
<span class="co">#&gt; Warning: Use of `df_toPlot$features` is discouraged. Use `features` instead.</span>
<span class="co">#&gt; Warning: Use of `df_toPlot$importance` is discouraged. Use `importance` instead.</span>
<span class="co">#&gt; Warning: Use of `df_toPlot$features` is discouraged. Use `features` instead.</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-30-1.png" width="768"></p>
<div class="sourceCode" id="cb38"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visImportance.html">visImportance</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, plot <span class="op">=</span> <span class="st">"heatmap"</span><span class="op">)</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-30-2.png" width="768"></p>
<div class="sourceCode" id="cb39"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="fu">metadata</span><span class="op">(</span><span class="va">sce_citeseq</span><span class="op">)</span><span class="op">[[</span><span class="st">"importanceADT"</span><span class="op">]</span><span class="op">]</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">]</span>
<span class="co">#&gt;               CD4               CD8              CD27               CD5 </span>
<span class="co">#&gt;         30.558333         29.224077         27.691326         22.097058 </span>
<span class="co">#&gt;               CD7      PECAM (CD31)             CD11b               CD2 </span>
<span class="co">#&gt;         16.041289         13.495346         12.716874         12.241439 </span>
<span class="co">#&gt;    MHCII (HLA-DR)              CD28              CD44 IL7Ralpha (CD127) </span>
<span class="co">#&gt;         10.332952         10.189411          9.006814          8.808518 </span>
<span class="co">#&gt;         HLA-A,B,C             CD62L             CD11c            CD45RA </span>
<span class="co">#&gt;          6.961884          5.921209          5.678501          5.612525 </span>
<span class="co">#&gt;      CD366 (tim3)               CD3              CD56      PD-1 (CD279) </span>
<span class="co">#&gt;          5.294986          5.263878          4.732472          4.427862</span></pre></div>
<p>The importance scores can be visualised in a boxplot and heatmap. Our evaluation of ADT importance show that unsurprisingly CD4 and CD8 are the top two discriminating proteins in PBMCs.</p>
<p>Let us try clustering with only ADTs with a score greater than 5.</p>
<div class="sourceCode" id="cb40"><pre class="downlit">
<span class="va">subset_adt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="fu">metadata</span><span class="op">(</span><span class="va">sce_citeseq</span><span class="op">)</span><span class="op">[[</span><span class="st">"importanceADT"</span><span class="op">]</span><span class="op">]</span> <span class="op">&gt;</span> <span class="fl">5</span><span class="op">)</span><span class="op">)</span>
<span class="va">subset_adt</span>
<span class="co">#&gt;  [1] "CD11b"             "CD11c"             "CD2"              </span>
<span class="co">#&gt;  [4] "CD27"              "CD28"              "CD3"              </span>
<span class="co">#&gt;  [7] "CD366 (tim3)"      "CD4"               "CD44"             </span>
<span class="co">#&gt; [10] "CD45RA"            "CD5"               "CD62L"            </span>
<span class="co">#&gt; [13] "CD7"               "CD8"               "HLA-A,B,C"        </span>
<span class="co">#&gt; [16] "IL7Ralpha (CD127)" "MHCII (HLA-DR)"    "PECAM (CD31)"</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span><span class="op">(</span><span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/CiteFuse.html">CiteFuse</a></span><span class="op">(</span><span class="va">sce_citeseq</span>,
                                    ADT_subset <span class="op">=</span> <span class="va">subset_adt</span>,
                                    metadata_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"W_SNF_adtSubset1"</span>,
                                                       <span class="st">"W_ADT_adtSubset1"</span>,
                                                       <span class="st">"W_RNA"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Calculating affinity matrix </span>
<span class="co">#&gt; Performing SNF</span>
<span class="co">#&gt;    user  system elapsed </span>
<span class="co">#&gt;   6.161   0.280   6.978</span>
<span class="va">SNF_W_clust_adtSubset1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/spectralClustering.html">spectralClustering</a></span><span class="op">(</span><span class="fu">metadata</span><span class="op">(</span><span class="va">sce_citeseq</span><span class="op">)</span><span class="op">[[</span><span class="st">"W_SNF_adtSubset1"</span><span class="op">]</span><span class="op">]</span>, K <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>
<span class="co">#&gt; Computing Spectral Clustering </span>
<span class="co">#&gt; Computing Diffusion Coordinates</span>
<span class="co">#&gt; Used default value: 5 dimensions</span>
<span class="va">sce_citeseq</span><span class="op">$</span><span class="va">SNF_W_clust_adtSubset1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">SNF_W_clust_adtSubset1</span><span class="op">$</span><span class="va">labels</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://mclust-org.github.io/mclust/">mclust</a></span><span class="op">)</span>
<span class="co">#&gt; Package 'mclust' version 5.4.6</span>
<span class="co">#&gt; Type 'citation("mclust")' for citing this R package in publications.</span>
<span class="fu"><a href="https://mclust-org.github.io/mclust/reference/adjustedRandIndex.html">adjustedRandIndex</a></span><span class="op">(</span><span class="va">sce_citeseq</span><span class="op">$</span><span class="va">SNF_W_clust_adtSubset1</span>, <span class="va">sce_citeseq</span><span class="op">$</span><span class="va">SNF_W_clust</span><span class="op">)</span>
<span class="co">#&gt; [1] 0.9305236</span></pre></div>
<p>When we compare between the two clustering outcomes, we find that the adjusted rand index is approximately 0.93, where a value of 1 denotes complete concordance.</p>
</div>
<div id="gene---adt-network" class="section level2">
<h2 class="hasAnchor">
<a href="#gene---adt-network" class="anchor"></a>Gene - ADT network</h2>
<p>The <code>geneADTnetwork</code> function plots an interaction network between genes identified from the DE analysis. The nodes denote proteins and RNA whilst the edges denote positive and negative correlation in expression.</p>
<div class="sourceCode" id="cb41"><pre class="downlit">
<span class="va">RNA_feature_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">rna_DEgenes_all</span>, <span class="st">"[["</span>, <span class="st">"name"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">ADT_feature_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">adt_DEgenes_all</span>, <span class="st">"[["</span>, <span class="st">"name"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/geneADTnetwork.html">geneADTnetwork</a></span><span class="op">(</span><span class="va">sce_citeseq</span>,
               RNA_feature_subset <span class="op">=</span> <span class="va">RNA_feature_subset</span>,
               ADT_feature_subset <span class="op">=</span> <span class="va">ADT_feature_subset</span>,
               cor_method <span class="op">=</span> <span class="st">"pearson"</span>,
               network_layout <span class="op">=</span> <span class="fu">igraph</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/igraph/man/layout_with_fr.html">layout_with_fr</a></span><span class="op">)</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-32-1.png" width="768"></p>
<pre><code>#&gt; IGRAPH 58acbaf UN-B 72 134 -- 
#&gt; + attr: name (v/c), label (v/c), class (v/c), type (v/l), shape (v/c),
#&gt; | color (v/c), size (v/n), label.cex (v/n), label.color (v/c), value
#&gt; | (e/n), color (e/c), weights (e/n)
#&gt; + edges from 58acbaf (vertex names):
#&gt;  [1] RNA_hg19_IL7R  --ADT_CD28             
#&gt;  [2] RNA_hg19_LTB   --ADT_CD28             
#&gt;  [3] RNA_hg19_NKG7  --ADT_CD28             
#&gt;  [4] RNA_hg19_CST7  --ADT_CD28             
#&gt;  [5] RNA_hg19_GNLY  --ADT_CD28             
#&gt;  [6] RNA_hg19_CCL5  --ADT_CD28             
#&gt; + ... omitted several edges</code></pre>
</div>
<div id="rna-ligand---adt-receptor-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#rna-ligand---adt-receptor-analysis" class="anchor"></a>RNA Ligand - ADT Receptor Analysis</h2>
<p>With the advent of CITE-seq, we can now predict ligand-receptor interactions by using cell surface protein expression. CiteFuse implements a <code>ligandReceptorTest</code> to find ligand receptor interactions between sender and receiver cells. Importantly, the ADT count is used to predict receptor expression within receiver cells. Note that the setting <code>altExp_name = "RNA"</code> would enable users to predict ligand-receptor interaction from RNA expression only.</p>
<div class="sourceCode" id="cb43"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"lr_pair_subset"</span>, package <span class="op">=</span> <span class="st">"CiteFuse"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">lr_pair_subset</span><span class="op">)</span>
<span class="co">#&gt;      [,1]          [,2]   </span>
<span class="co">#&gt; [1,] "hg19_IL17RA" "CD45" </span>
<span class="co">#&gt; [2,] "hg19_FAS"    "CD11b"</span>
<span class="co">#&gt; [3,] "hg19_GZMK"   "CD62L"</span>
<span class="co">#&gt; [4,] "hg19_CD40LG" "CD11b"</span>
<span class="co">#&gt; [5,] "hg19_FLT3LG" "CD62L"</span>
<span class="co">#&gt; [6,] "hg19_GZMA"   "CD19"</span>
<span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/normaliseExprs.html">normaliseExprs</a></span><span class="op">(</span>sce <span class="op">=</span> <span class="va">sce_citeseq</span>, 
                              altExp_name <span class="op">=</span> <span class="st">"ADT"</span>, 
                              transform <span class="op">=</span> <span class="st">"zi_minMax"</span><span class="op">)</span>
<span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/normaliseExprs.html">normaliseExprs</a></span><span class="op">(</span>sce <span class="op">=</span> <span class="va">sce_citeseq</span>, 
                              altExp_name <span class="op">=</span> <span class="st">"none"</span>, 
                              exprs_value <span class="op">=</span> <span class="st">"logcounts"</span>,
                              transform <span class="op">=</span> <span class="st">"minMax"</span><span class="op">)</span>
<span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/ligandReceptorTest.html">ligandReceptorTest</a></span><span class="op">(</span>sce <span class="op">=</span> <span class="va">sce_citeseq</span>,
                                  ligandReceptor_list <span class="op">=</span> <span class="va">lr_pair_subset</span>,
                                  cluster <span class="op">=</span> <span class="va">sce_citeseq</span><span class="op">$</span><span class="va">SNF_W_louvain</span>,
                                  RNA_exprs_value <span class="op">=</span> <span class="st">"minMax"</span>,
                                  use_alt_exp <span class="op">=</span> <span class="cn">TRUE</span>,
                                  altExp_name <span class="op">=</span> <span class="st">"ADT"</span>,
                                  altExp_exprs_value <span class="op">=</span> <span class="st">"zi_minMax"</span>,
                                  num_permute <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span> 
<span class="co">#&gt; 100 ......200 ......300 ......400 ......500 ......600 ......700 ......800 ......900 ......1000 ......</span></pre></div>
<div class="sourceCode" id="cb44"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visLigandReceptor.html">visLigandReceptor</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, 
                  type <span class="op">=</span> <span class="st">"pval_heatmap"</span>,
                  receptor_type <span class="op">=</span> <span class="st">"ADT"</span><span class="op">)</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-34-1.png" width="768"></p>
<div class="sourceCode" id="cb45"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visLigandReceptor.html">visLigandReceptor</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, 
                  type <span class="op">=</span> <span class="st">"pval_dotplot"</span>,
                  receptor_type <span class="op">=</span> <span class="st">"ADT"</span><span class="op">)</span>
<span class="co">#&gt; Warning: Use of `df_toPlot$value` is discouraged. Use `value` instead.</span>

<span class="co">#&gt; Warning: Use of `df_toPlot$value` is discouraged. Use `value` instead.</span>
<span class="co">#&gt; Warning: Use of `df_toPlot$Var1` is discouraged. Use `Var1` instead.</span>
<span class="co">#&gt; Warning: Use of `df_toPlot$Var2` is discouraged. Use `Var2` instead.</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-35-1.png" width="576"></p>
<div class="sourceCode" id="cb46"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visLigandReceptor.html">visLigandReceptor</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, 
                  type <span class="op">=</span> <span class="st">"group_network"</span>,
                  receptor_type <span class="op">=</span> <span class="st">"ADT"</span><span class="op">)</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-36-1.png" width="768"></p>
<div class="sourceCode" id="cb47"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visLigandReceptor.html">visLigandReceptor</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, 
                  type <span class="op">=</span> <span class="st">"group_heatmap"</span>,
                  receptor_type <span class="op">=</span> <span class="st">"ADT"</span><span class="op">)</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-37-1.png" width="768"></p>
<div class="sourceCode" id="cb48"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visLigandReceptor.html">visLigandReceptor</a></span><span class="op">(</span><span class="va">sce_citeseq</span>,
                  type <span class="op">=</span> <span class="st">"lr_network"</span>,
                  receptor_type <span class="op">=</span> <span class="st">"ADT"</span><span class="op">)</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-38-1.png" width="768"></p>
</div>
<div id="between-sample-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#between-sample-analysis" class="anchor"></a>Between-sample analysis</h2>
<p>Lastly, we will jointly analyse the current PBMC CITE-seq data, taken from healthy human donors, and another subset of CITE-seq data from patients with cutaneous T-cell lymphoma (CTCL), again from Mimitou et al. (2019). The data <code>sce_ctcl_subset</code> provided in our <code>CiteFuse</code> package already contains the clustering information.</p>
<div class="sourceCode" id="cb49"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"sce_ctcl_subset"</span>, package <span class="op">=</span> <span class="st">"CiteFuse"</span><span class="op">)</span></pre></div>
<p>To visualise and compare gene or protein expression data, we can use <code>visualiseExprsList</code> function.</p>
<div class="sourceCode" id="cb50"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visualiseExprsList.html">visualiseExprsList</a></span><span class="op">(</span>sce_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>control <span class="op">=</span> <span class="va">sce_citeseq</span>,
                                   ctcl <span class="op">=</span> <span class="va">sce_ctcl_subset</span><span class="op">)</span>,
                   plot <span class="op">=</span> <span class="st">"boxplot"</span>,
                   altExp_name <span class="op">=</span> <span class="st">"none"</span>,
                   exprs_value <span class="op">=</span> <span class="st">"logcounts"</span>,
                   feature_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"hg19_S100A10"</span>, <span class="st">"hg19_CD8A"</span><span class="op">)</span>,
                   group_by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"SNF_W_louvain"</span>, <span class="st">"SNF_W_louvain"</span><span class="op">)</span><span class="op">)</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-40-1.png" width="700"></p>
<div class="sourceCode" id="cb51"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/visualiseExprsList.html">visualiseExprsList</a></span><span class="op">(</span>sce_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>control <span class="op">=</span> <span class="va">sce_citeseq</span>,
                                   ctcl <span class="op">=</span> <span class="va">sce_ctcl_subset</span><span class="op">)</span>,
                   plot <span class="op">=</span> <span class="st">"boxplot"</span>,
                   altExp_name <span class="op">=</span> <span class="st">"ADT"</span>, 
                   feature_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"CD19"</span>, <span class="st">"CD8"</span><span class="op">)</span>,
                   group_by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"SNF_W_louvain"</span>, <span class="st">"SNF_W_louvain"</span><span class="op">)</span><span class="op">)</span></pre></div>
<p><img src="CiteFuse_BioCAsia_workshop_files/figure-html/unnamed-chunk-40-2.png" width="700"></p>
<p>We can then perform differential expression analysis of the RNA expression level across the two clusters that have high CD19 expression in ADT.</p>
<div class="sourceCode" id="cb52"><pre class="downlit">
<span class="va">de_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/DEgenesCross.html">DEgenesCross</a></span><span class="op">(</span>sce_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>control <span class="op">=</span> <span class="va">sce_citeseq</span>,
                                       ctcl <span class="op">=</span> <span class="va">sce_ctcl_subset</span><span class="op">)</span>,
                       colData_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"SNF_W_louvain"</span>, <span class="st">"SNF_W_louvain"</span><span class="op">)</span>,
                       group_to_test <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"2"</span>, <span class="st">"6"</span><span class="op">)</span><span class="op">)</span>
<span class="va">de_res_filter</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/selectDEgenes.html">selectDEgenes</a></span><span class="op">(</span>de_res <span class="op">=</span> <span class="va">de_res</span><span class="op">)</span>
<span class="va">de_res_filter</span>
<span class="co">#&gt; $control</span>
<span class="co">#&gt;            stats.W         pval     p.adjust meanExprs.1 meanExprs.2  meanPct.1</span>
<span class="co">#&gt; hg19_NKG7     82.0 9.717299e-13 9.717299e-13  0.51066894   3.2333024 0.30232558</span>
<span class="co">#&gt; hg19_CST7    138.0 5.875063e-12 5.875063e-12  0.22512314   2.2146495 0.13953488</span>
<span class="co">#&gt; hg19_GNLY    186.0 8.935760e-12 8.935760e-12  0.07220401   2.7782587 0.02325581</span>
<span class="co">#&gt; hg19_CCL5    124.0 2.418174e-11 2.418174e-11  1.04348029   3.7809281 0.37209302</span>
<span class="co">#&gt; hg19_GZMH    258.0 1.807873e-10 1.807873e-10  0.00000000   1.4711399 0.00000000</span>
<span class="co">#&gt; hg19_ITGB2   210.0 4.203966e-09 4.203966e-09  0.44185664   1.7346818 0.44186047</span>
<span class="co">#&gt; hg19_KLRD1   322.5 5.000482e-08 5.000482e-08  0.09327555   1.0358615 0.09302326</span>
<span class="co">#&gt; hg19_GZMA    284.5 5.732252e-08 5.732252e-08  0.32515702   1.5433119 0.18604651</span>
<span class="co">#&gt; hg19_ZEB2    387.0 7.878997e-08 7.878997e-08  0.00000000   0.8211807 0.00000000</span>
<span class="co">#&gt; hg19_CD99    274.0 1.679625e-07 1.679625e-07  0.58551572   1.5699157 0.46511628</span>
<span class="co">#&gt;            meanPct.2  meanDiff   pctDiff       name   group</span>
<span class="co">#&gt; hg19_NKG7  1.0000000 0.6976744 0.6976744  hg19_NKG7 control</span>
<span class="co">#&gt; hg19_CST7  0.8947368 0.7552020 0.7552020  hg19_CST7 control</span>
<span class="co">#&gt; hg19_GNLY  0.7894737 0.7662179 0.7662179  hg19_GNLY control</span>
<span class="co">#&gt; hg19_CCL5  1.0000000 0.6279070 0.6279070  hg19_CCL5 control</span>
<span class="co">#&gt; hg19_GZMH  0.6842105 0.6842105 0.6842105  hg19_GZMH control</span>
<span class="co">#&gt; hg19_ITGB2 0.8684211 0.4265606 0.4265606 hg19_ITGB2 control</span>
<span class="co">#&gt; hg19_KLRD1 0.6578947 0.5648715 0.5648715 hg19_KLRD1 control</span>
<span class="co">#&gt; hg19_GZMA  0.8157895 0.6297430 0.6297430  hg19_GZMA control</span>
<span class="co">#&gt; hg19_ZEB2  0.5263158 0.5263158 0.5263158  hg19_ZEB2 control</span>
<span class="co">#&gt; hg19_CD99  0.8947368 0.4296206 0.4296206  hg19_CD99 control</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $ctcl</span>
<span class="co">#&gt;                stats.W         pval     p.adjust meanExprs.1 meanExprs.2</span>
<span class="co">#&gt; hg19_RPS26         0.0 1.087530e-14 1.087530e-14  1.42819970    5.072471</span>
<span class="co">#&gt; hg19_LTB         235.0 1.478865e-08 1.478865e-08  0.48566164    1.978730</span>
<span class="co">#&gt; hg19_SELL        298.5 3.618421e-08 3.618421e-08  0.10674144    1.152572</span>
<span class="co">#&gt; hg19_LEPROTL1    268.0 9.163954e-08 9.163954e-08  0.37427389    1.300745</span>
<span class="co">#&gt; hg19_NOSIP       306.0 2.589055e-07 2.589055e-07  0.23412389    1.157235</span>
<span class="co">#&gt; hg19_LINC02446   352.0 3.887845e-07 3.887845e-07  0.09889831    1.004083</span>
<span class="co">#&gt; hg19_FOS         353.5 5.438707e-07 5.438707e-07  0.14088922    1.125337</span>
<span class="co">#&gt; hg19_TMEM123     331.0 1.090629e-06 1.090629e-06  0.26112624    1.012034</span>
<span class="co">#&gt; hg19_LDHB        325.0 2.344648e-06 2.344648e-06  0.70148411    1.588459</span>
<span class="co">#&gt; hg19_NPM1        323.0 2.836125e-06 2.836125e-06  1.24903807    2.202268</span>
<span class="co">#&gt;                 meanPct.1 meanPct.2  meanDiff   pctDiff           name group</span>
<span class="co">#&gt; hg19_RPS26     0.84210526 1.0000000 0.1578947 0.1578947     hg19_RPS26  ctcl</span>
<span class="co">#&gt; hg19_LTB       0.28947368 0.9069767 0.6175031 0.6175031       hg19_LTB  ctcl</span>
<span class="co">#&gt; hg19_SELL      0.07894737 0.6976744 0.6187271 0.6187271      hg19_SELL  ctcl</span>
<span class="co">#&gt; hg19_LEPROTL1  0.28947368 0.9069767 0.6175031 0.6175031  hg19_LEPROTL1  ctcl</span>
<span class="co">#&gt; hg19_NOSIP     0.21052632 0.7674419 0.5569155 0.5569155     hg19_NOSIP  ctcl</span>
<span class="co">#&gt; hg19_LINC02446 0.07894737 0.6279070 0.5489596 0.5489596 hg19_LINC02446  ctcl</span>
<span class="co">#&gt; hg19_FOS       0.07894737 0.6511628 0.5722154 0.5722154       hg19_FOS  ctcl</span>
<span class="co">#&gt; hg19_TMEM123   0.23684211 0.7674419 0.5305998 0.5305998   hg19_TMEM123  ctcl</span>
<span class="co">#&gt; hg19_LDHB      0.50000000 0.8604651 0.3604651 0.3604651      hg19_LDHB  ctcl</span>
<span class="co">#&gt; hg19_NPM1      0.68421053 0.9534884 0.2692778 0.2692778      hg19_NPM1  ctcl</span></pre></div>
</div>
<div id="application-of-citefuse-on-asap-seq" class="section level2">
<h2 class="hasAnchor">
<a href="#application-of-citefuse-on-asap-seq" class="anchor"></a>Application of CiteFuse on ASAP-seq</h2>
</div>
</div>
<div id="final-notes" class="section level1">
<h1 class="hasAnchor">
<a href="#final-notes" class="anchor"></a>Final notes</h1>
<p>A compiled version of the vignette is available on the <a href="https://github.com/SydneyBioX/BiocAsia2020CiteFuse/articles/workshopTCiteFuse.html">workshop website</a>.</p>
<p>If you have questions that you could not ask during the workshop, feel free to open an issue on the github repository <a href="https://github.com/SydneyBioX/BiocAsia2020CiteFuse/issues">here</a>.</p>
<div class="sourceCode" id="cb53"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.0.2 (2020-06-22)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Mojave 10.14.6</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] parallel  stats4    stats     graphics  grDevices utils     datasets </span>
<span class="co">#&gt; [8] methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] mclust_5.4.6                gridExtra_2.3              </span>
<span class="co">#&gt;  [3] DT_0.14                     scater_1.16.2              </span>
<span class="co">#&gt;  [5] ggplot2_3.3.2               SingleCellExperiment_1.10.1</span>
<span class="co">#&gt;  [7] SummarizedExperiment_1.18.1 DelayedArray_0.14.0        </span>
<span class="co">#&gt;  [9] matrixStats_0.56.0          Biobase_2.48.0             </span>
<span class="co">#&gt; [11] GenomicRanges_1.40.0        GenomeInfoDb_1.24.2        </span>
<span class="co">#&gt; [13] IRanges_2.22.2              S4Vectors_0.26.1           </span>
<span class="co">#&gt; [15] BiocGenerics_0.34.0         CiteFuse_1.0.0             </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] Rtsne_0.15                ggbeeswarm_0.6.0         </span>
<span class="co">#&gt;   [3] colorspace_1.4-1          ellipsis_0.3.1           </span>
<span class="co">#&gt;   [5] ggridges_0.5.2            rprojroot_1.3-2          </span>
<span class="co">#&gt;   [7] XVector_0.28.0            BiocNeighbors_1.6.0      </span>
<span class="co">#&gt;   [9] fs_1.4.2                  rstudioapi_0.11          </span>
<span class="co">#&gt;  [11] farver_2.0.3              graphlayouts_0.7.0       </span>
<span class="co">#&gt;  [13] ggrepel_0.8.2             RSpectra_0.16-0          </span>
<span class="co">#&gt;  [15] splines_4.0.2             knitr_1.29               </span>
<span class="co">#&gt;  [17] heatmap.plus_1.3          polyclip_1.10-0          </span>
<span class="co">#&gt;  [19] jsonlite_1.7.0            alluvial_0.1-2           </span>
<span class="co">#&gt;  [21] kernlab_0.9-29            pheatmap_1.0.12          </span>
<span class="co">#&gt;  [23] uwot_0.1.8                ggforce_0.3.2            </span>
<span class="co">#&gt;  [25] ExPosition_2.8.23         compiler_4.0.2           </span>
<span class="co">#&gt;  [27] dqrng_0.2.1               prettyGraphs_2.1.6       </span>
<span class="co">#&gt;  [29] backports_1.1.8           assertthat_0.2.1         </span>
<span class="co">#&gt;  [31] Matrix_1.2-18             limma_3.44.3             </span>
<span class="co">#&gt;  [33] tweenr_1.0.1              BiocSingular_1.4.0       </span>
<span class="co">#&gt;  [35] htmltools_0.5.0           tools_4.0.2              </span>
<span class="co">#&gt;  [37] rsvd_1.0.3                igraph_1.2.5             </span>
<span class="co">#&gt;  [39] gtable_0.3.0              glue_1.4.1               </span>
<span class="co">#&gt;  [41] GenomeInfoDbData_1.2.3    reshape2_1.4.4           </span>
<span class="co">#&gt;  [43] dplyr_1.0.0               Rcpp_1.0.4.6             </span>
<span class="co">#&gt;  [45] pkgdown_1.6.1             vctrs_0.3.1              </span>
<span class="co">#&gt;  [47] crosstalk_1.1.0.1         DelayedMatrixStats_1.10.1</span>
<span class="co">#&gt;  [49] ggraph_2.0.3              xfun_0.15                </span>
<span class="co">#&gt;  [51] stringr_1.4.0             lifecycle_0.2.0          </span>
<span class="co">#&gt;  [53] irlba_2.3.3               statmod_1.4.34           </span>
<span class="co">#&gt;  [55] edgeR_3.30.3              zlibbioc_1.34.0          </span>
<span class="co">#&gt;  [57] MASS_7.3-51.6             scales_1.1.1             </span>
<span class="co">#&gt;  [59] tidygraph_1.2.0           ragg_0.3.1               </span>
<span class="co">#&gt;  [61] rhdf5_2.32.2              RColorBrewer_1.1-2       </span>
<span class="co">#&gt;  [63] SNFtool_2.3.0             yaml_2.2.1               </span>
<span class="co">#&gt;  [65] memoise_1.1.0             segmented_1.2-0          </span>
<span class="co">#&gt;  [67] stringi_1.4.6             desc_1.2.0               </span>
<span class="co">#&gt;  [69] randomForest_4.6-14       scran_1.16.0             </span>
<span class="co">#&gt;  [71] BiocParallel_1.22.0       rlang_0.4.6              </span>
<span class="co">#&gt;  [73] pkgconfig_2.0.3           systemfonts_0.3.2        </span>
<span class="co">#&gt;  [75] bitops_1.0-6              evaluate_0.14            </span>
<span class="co">#&gt;  [77] lattice_0.20-41           purrr_0.3.4              </span>
<span class="co">#&gt;  [79] Rhdf5lib_1.10.0           labeling_0.3             </span>
<span class="co">#&gt;  [81] htmlwidgets_1.5.1         cowplot_1.0.0            </span>
<span class="co">#&gt;  [83] tidyselect_1.1.0          plyr_1.8.6               </span>
<span class="co">#&gt;  [85] magrittr_1.5              R6_2.4.1                 </span>
<span class="co">#&gt;  [87] generics_0.0.2            pillar_1.4.4             </span>
<span class="co">#&gt;  [89] withr_2.2.0               mixtools_1.2.0           </span>
<span class="co">#&gt;  [91] survival_3.1-12           RCurl_1.98-1.2           </span>
<span class="co">#&gt;  [93] tibble_3.0.1              crayon_1.3.4             </span>
<span class="co">#&gt;  [95] rmarkdown_2.3             viridis_0.5.1            </span>
<span class="co">#&gt;  [97] locfit_1.5-9.4            grid_4.0.2               </span>
<span class="co">#&gt;  [99] FNN_1.1.3                 propr_4.2.6              </span>
<span class="co">#&gt; [101] digest_0.6.25             tidyr_1.1.0              </span>
<span class="co">#&gt; [103] dbscan_1.1-5              munsell_0.5.0            </span>
<span class="co">#&gt; [105] beeswarm_0.2.3            viridisLite_0.3.0        </span>
<span class="co">#&gt; [107] vipor_0.4.5</span></pre></div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p><a href="mailto:hani.kim@sydney.edu.au" class="email">hani.kim@sydney.edu.au</a><a href="#fnref1" class="footnote-back">↩</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Sean Davis.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
